src/Npgsql/NpgsqlConnector.cs

```diff
@@ -674,14 +674,23 @@ async Task RawOpen(NpgsqlTimeout timeout, bool async, CancellationToken cancella
                         break;
                     case 'S':
                         var clientCertificates = new X509Certificate2Collection();
-                        var certPath = Settings.ClientCertificate ?? PostgresEnvironment.SslCert;
-
-                        if (certPath is null && PostgresEnvironment.SslCertDefault is string certPathDefault)
-                            certPath = certPathDefault;
+                        var certPath = Settings.ClientCertificate ?? PostgresEnvironment.SslCert ?? PostgresEnvironment.SslCertDefault;
 
                         if (certPath != null)
                         {
-                            cert = new X509Certificate2(certPath, Settings.ClientCertificateKey ?? PostgresEnvironment.SslKey);
+                            var password = Settings.ClientCertificateKeyPassword;
+#if NET5_0
```

Comment:
Should we throw NotSupportedException for pre-.NET 5.0 if the certificate isn't pfx?

Reply:
Yeah, and how in the world I didn't think about that...